PolyEngine/Core/Src/EnumUtils.hpp

```diff
@@ -99,8 +97,14 @@ namespace Poly {
 		const T* GetData() const { return Data; }
 
 		//------------------------------------------------------------------------------
-		T& operator[](E idx) { HEAVY_ASSERTE(static_cast<IndexType>(idx) < GetSize(), "Index out of bounds!"); return Data[static_cast<IndexType>(idx)]; }
-		const T& operator[](E idx) const { HEAVY_ASSERTE(static_cast<IndexType>(idx) < GetSize(), "Index out of bounds!"); return Data[static_cast<IndexType>(idx)]; }
+		T& operator[](E enumValue) {
+			return const_cast<T&>(const_cast<const EnumArray&>(*this)[enumValue]); //as advised by Effective C++
```

Comment:
This is actually fully safe, as supported by Effective C++, Third Edition, by Scott Meyers and _extensive_ online research 😃